File: golang/internal/backendmqtt/Topics.go
package backendmqtt

import "fmt"

const (
	clientStatusTopicFormat     = "accounts/%s/clients/%s/status"
	clientWebRTCInboxTopicFormat = "accounts/%s/clients/%s/webrtc_inbox"
	clientControlInboxTopicFormat = "accounts/%s/clients/%s/control_inbox"
	babyStationsTopicFormat     = "accounts/%s/baby_stations"
	parentStationsTopicFormat   = "accounts/%s/parent_stations"
)

func ClientStatusTopic(accountID string, clientID string) string {
	return fmt.Sprintf(clientStatusTopicFormat, accountID, clientID)
}

func ClientWebRTCInboxTopic(accountID string, clientID string) string {
	return fmt.Sprintf(clientWebRTCInboxTopicFormat, accountID, clientID)
}

func ClientControlInboxTopic(accountID string, clientID string) string {
	return fmt.Sprintf(clientControlInboxTopicFormat, accountID, clientID)
}

func BabyStationsTopic(accountID string) string {
	return fmt.Sprintf(babyStationsTopicFormat, accountID)
}

func ParentStationsTopic(accountID string) string {
	return fmt.Sprintf(parentStationsTopicFormat, accountID)
}

File: frontend/src/services/MQTTService/topics.ts
export const clientStatusTopic = (accountID: string, clientID: string): string => {
    return `accounts/${accountID}/clients/${clientID}/status`;
};

export interface ClientStatusTopicParts {
    account_id: string;
    client_id: string;
}

export const parse_client_status_topic = (topic: string): Optional<ClientStatusTopicParts> => {
    const match = topic.match(/^accounts\/([^/]+)\/clients\/([^/]+)\/status$/);
    if (match === null) return null;
    return {
        account_id: match[1],
        client_id: match[2],
    };
};
